provider/azure.go

```diff
@@ -224,60 +222,58 @@ func (p *AzureProvider) ApplyChanges(ctx context.Context, changes *plan.Changes)
 }
 
 func (p *AzureProvider) zones(ctx context.Context) ([]dns.Zone, error) {
-	log.Debug("Retrieving Azure DNS zones.")
+	log.Debugf("Retrieving Azure DNS zones for rg: %s.", p.resourceGroup)
 
 	var zones []dns.Zone
-	list, err := p.zonesClient.ListByResourceGroup(ctx, p.resourceGroup, nil)
+
+	i, err := p.zonesClient.ListByResourceGroupComplete(ctx, p.resourceGroup, nil)
+
 	if err != nil {
 		return nil, err
 	}
 
-	for list.Values() != nil && len(list.Values()) > 0 {
-		for _, zone := range list.Values() {
-			if zone.Name == nil {
-				continue
-			}
+	for i.NotDone() {
+		zone := i.Value()
+		log.Debugf("Validating Zone: %v", *zone.Name)
 
-			if !p.domainFilter.Match(*zone.Name) {
-				continue
-			}
+		if zone.Name == nil {
+			continue
+		}
 
-			if !p.zoneIDFilter.Match(*zone.ID) {
-				continue
-			}
+		if !p.domainFilter.Match(*zone.Name) {
+			continue
+		}
 
-			zones = append(zones, zone)
+		if !p.zoneIDFilter.Match(*zone.ID) {
+			continue
 		}
 
-		//list, err = p.zonesClient.listByResourceGroupNextResults(ctx, list)
-		//if err != nil {
-		//	return nil, err
-		//}
+		zones = append(zones, zone)
+
+		i.NextWithContext(ctx)
 	}
+
 	log.Debugf("Found %d Azure DNS zone(s).", len(zones))
 	return zones, nil
 }
 
 func (p *AzureProvider) iterateRecords(ctx context.Context, zoneName string, callback func(dns.RecordSet) bool) error {
 	log.Debugf("Retrieving Azure DNS records for zone '%s'.", zoneName)
 
-	list, err := p.recordSetsClient.ListByDNSZone(ctx, p.resourceGroup, zoneName, nil, "") // TODO
+	i, err := p.recordSetsClient.ListAllByDNSZoneComplete(ctx, p.resourceGroup, zoneName, nil, "")
+
 	if err != nil {
 		return err
 	}
 
-	for list.Values() != nil && len(list.Values()) > 0 {
-		for _, recordSet := range list.Values() {
-			if !callback(recordSet) {
-				return nil
-			}
+	for i.NotDone() {
```

Comment:
this is currently returning done, with no pages in the tests, haven't managed to figure out why yet